packages/discovery-react-components/src/utils/document/documentUtils.ts

```diff
@@ -99,13 +89,26 @@ export function getTextNodeAndOffset(node: Node, offset: number): NodeOffset {
     // html entities (eg. `&quot;`), however those entities get rendered in
     // the dom which throws off the offsets values, for example `&quot;`
     // becomes `"`, going from 6 characters to 1.
-    const encodedTextSubstring = originalText.substring(0, textOffset);
+    const encodedTextSubstring = originalText.substring(0, domOffset);
     const decodedText = decodeHTML(encodedTextSubstring);
-    const adjustment = textOffset - decodedText.length;
-    textOffset = Math.max(0, textOffset - adjustment);
+    const adjustment = domOffset - decodedText.length;
+    domOffset = Math.max(0, domOffset - adjustment);
+  }
+
+  // In certain cases (i.e. text length > 65536 chars), the DOM will break
+  // up text into multiple TextNodes. Loop through TextNodes, finding which
+  // contains the `domOffset`
+  const iterator = document.createNodeIterator(node, NodeFilter.SHOW_TEXT);
+  let textNode: Text, len: number;
+  do {
+    textNode = iterator.nextNode() as Text;
+  } while (textNode && (len = textNode.data.length) && domOffset > len && (domOffset -= len));
```

Comment:
I still don't love the assignments being done in the `while` condition, but those have been around for a while...

Reply:
It's the best I could think of where the initialization happens both before and during, and the checks need to happen after that. Suggestions for improvement welcome.

Reply:
That's fair, any way I can think of to do it differently would lead to less clean code.